lib/snowpack.ex

```diff
@@ -76,32 +95,39 @@ defmodule Snowpack do
 
   See `DBConnection.start_link/2` for more information and a full list of available options.
 
+  ### Library Options
+
+  The given options are used to configure snowpack:
+
+    * `:event_prefix` - Used to prefix `Telemetry` events (required)
+
   ## Examples
 
-  Start connection using the default configuration (UNIX domain socket):
+  Start connection using basic User / Pass configuration:
+
+      iex> {:ok, pid} = Snowpack.start_link(event_prefix: :my_app, connection: [server: "account-id.snowflakecomputing.com", uid: "USER", pwd: "PASS"])
+      {:ok, #PID<0.69.0>}
+
+  Start connection using DNS configuration:
 
-      iex> {:ok, pid} = Snowpack.start_link(connection: [server: "account-id.snowflakecomputing.com", uid: "USER", pwd: "PASS"])
+      iex> {:ok, pid} = Snowpack.start_link(event_prefix: :my_app, connection: [dsn: "snowflake"])
       {:ok, #PID<0.69.0>}
 
   Run a query after connection has been established:
 
-      iex> {:ok, pid} = Snowpack.start_link(after_connect: &Snowpack.query!(&1, "SET time_zone = '+00:00'"))
+      iex> {:ok, pid} = Snowpack.start_link(event_prefix: :my_app, connection: [dsn: "snowflake"], after_connect: &Snowpack.query!(&1, "SET time_zone = '+00:00'"))
       {:ok, #PID<0.69.0>}
 
   """
   @spec start_link([start_option()]) :: {:ok, pid()} | {:error, Snowpack.Error.t()}
-  def start_link(options) do
-    options = Keyword.put_new(options, :idle_interval, @default_session_keepalive)
+  def start_link(opts) do
+    _ = Keyword.get(opts, :event_prefix) || raise ArgumentError, "must supply a event_prefix"
```

Comment:
The idea behind `event_prefix` is so that you can distinguish between multiple instances of `Snowpack` configured for a single app. Similar to how `Ecto` events are prefixed by the `repo` name since you can have multiple `Ecto` repos in a single app.